src/scenic/core/simulators.py

```diff
@@ -247,7 +247,7 @@ def currentState(self):
 
         The default implementation returns a tuple of the positions of all objects.
         """
-        return tuple(obj.position for obj in self.objects)
+        return {obj.name:[obj.position, obj.heading] for obj in self.objects}
```

Comment:
Changing this default behavior would break compatibility with people who are assuming it behaves as it did before (and as described in the docstring). It's probably a good idea to save the heading in `trajectory` as well, but I'd like to wait until the next backwards-incompatible release of Scenic.

In the mean time, we now have the `record` statement for saving additional information from the simulation; I'd recommend using that, or implementing your own subclass of `CarlaSimulation` that overrides `currentState`.

Reply:
I probably included this change accidentally because I wanted to focus only on one feature for this PR. Thanks for pointing out the alternative options, `record` and sub-classing.